python/cudf/cudf/core/buffer/exposure_tracked_buffer.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+import weakref
+from typing import (
+    Any,
+    Container,
+    Literal,
+    Mapping,
+    Optional,
+    Type,
+    TypeVar,
+    cast,
+)
+
+from typing_extensions import Self
+
+import cudf
+from cudf.core.buffer.buffer import Buffer, get_ptr_and_size
+from cudf.utils.string import format_bytes
+
+T = TypeVar("T", bound="ExposureTrackedBuffer")
+
+
+def get_owner(data, klass: Type[T]) -> Optional[T]:
```

Comment:
This function seems a bit odd. If `klass` is None then `get_owner` is always `None`? Is this functionality going to change once SpillableBuffer goes away and spilling relies directly on ExposureTrackedBuffer? Or is it going to stick around this way because of slices?

Reply:
OK from reading below it looks like this is necessary because we'll always be working with BufferSlice objects.